pwiz_tools/Skyline/SkylineFiles.cs

```diff
@@ -894,6 +895,146 @@ private bool CheckResults(SrmDocument document, string path, FormEx parent)
             return true;
         }
 
+        private void openPanorama_Click(object sender, EventArgs e)
+        {
+            OpenFromPanorama();
+        }
+
+        /// <summary>
+        /// Method used for testing server folder browser
+        /// </summary>
+        /// <param name="server"></param>
+        /// <param name="user"></param>
+        /// <param name="pass"></param>
+        /// <param name="folderJson"></param>
+        /// <param name="fileJson"></param>
+        public void OpenFromPanorama(string server, string user, string pass, JToken folderJson, JToken fileJson = null)
+        {
+            var panoramaClient = new WebPanoramaClient(new Uri(server));
+            using var dlg = new FilePicker();
+            dlg.InitializeTestDialog(new Uri(server), user, pass, folderJson, fileJson);
+            if (dlg.ShowDialog() != DialogResult.Cancel)
+            {
+
+                var folderPath = string.Empty;
+                if (!string.IsNullOrEmpty(Settings.Default.LastFolderPath))
+                {
+                    folderPath = Settings.Default.LastFolderPath;
+                }
+                
+            }
+        }
+
+        public void OpenFromPanorama()
+        {
+            var servers = Settings.Default.ServerList;
+            if (servers.Count == 0)
+            {
+                DialogResult buttonPress = MultiButtonMsgDlg.Show(
+                    this,
+                    TextUtil.LineSeparate(
+                        Resources.SkylineWindow_ShowPublishDlg_There_are_no_Panorama_servers_to_upload_to,

```

Comment:
This message has to be different.  It can't say "There are no Panorama servers to upload to".  We also need to be able to add a server without providing a username and password.  I will take a look at that.

Reply:
What do you think a better message would be? I changed it to say 'There are no Panorama servers to download from' although maybe it should be more general like 'There are no Panorama servers to connect to'?